app/src/main/java/com/example/yard/activities/PostActivity.java

```diff
@@ -143,14 +149,22 @@ public void onRequestPermissionsResult(
     super.onRequestPermissionsResult(requestCode, permissions, grantResults);
   }
 
-  private void launchcamera() {
-    Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
-    photoFile = getPhotoFileUri(PHOTO_FILE_NAME);
-    Uri fileProvider =
-        FileProvider.getUriForFile(PostActivity.this, "com.codepath.fileprovider", photoFile);
-    intent.putExtra(MediaStore.EXTRA_OUTPUT, fileProvider);
-    if (intent.resolveActivity(PostActivity.this.getPackageManager()) != null) {
-      startActivityForResult(intent, CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE);
+  private void uploadImage(Boolean takePicture) {
```

Comment:
No need to reuse this function if taking picture and uploading image almost don't have shared business logics. For example, there's no lines are shared between the 2 conditions.